scipy/stats/_continuous_distns.py

```diff
@@ -4707,6 +4707,42 @@ def _entropy(self):
         # https://en.wikipedia.org/wiki/Logistic_distribution
         return 2.0
 
+    def fit(self, data, *args, **kwds):
+        data, floc, fscale = _check_fit_input_parameters(self, data,
+                                                         args, kwds)
+
+        # if user has provided `floc` or `fscale`, fall back on super fit
+        # method. This scenario is not suitable for solving a system of
+        # equations
+        if floc is not None or fscale is not None:
+            return super(logistic_gen, self).fit(data, *args, **kwds)
+
+        # rv_continuous provided guesses
+        loc, scale = self._fitstart(data)
+        # account for user provided guesses
+        loc = kwds.pop('loc', loc)
+        scale = kwds.pop('scale', scale)
+
+        # the maximum likelihood estimators `a` and `b` of the location and
+        # scale parameters are roots of the two equations described in `func`.
+        # Source: Statistical Distributions, 3rd Edition. Evans, Hastings, and
+        # Peacock (2000), Page 130
+        def func(input, data):
+            a, b = input
+            data = np.asarray(data)
+            n = len(data)
+
+            # use technique from `scipy.special.logsumexp` to avoid overflow
+            c = (data - a) / b
+            M = c.max()
+            exp_l = (np.exp(M) * np.exp(c - M))
+
+            x1 = np.sum(exp_l / (1 + exp_l)) - (n / 2)
+            x2 = np.sum((c * ((exp_l - 1) / (exp_l + 1)))) - n
+            return x1, x2
```

Comment:
@swallan, in this case, factoring out `exp(M)` doesn't really help, because we still calculate `exp(M)`, and that could overflow.

There is something I should have noticed the first time I looked at this:  the expressions that involve the exponentials can be written using existing functions, and these will avoid the possibility of overflow.  In particular, `exp(c)/(1 + exp(c))` is the same as `scipy.special.expit(c)`, and `(exp(c) - 1)/(exp(c) + 1)` is `tanh(c/2)`.

Two more worthwhile changes: don't use `input` as a parameter name; and `data` was already passed through `asarray()` in `_check_fit_input_parameters`, so we don't need to do it again each time `func` is called.

Combining all these suggestions gives
```suggestion
        def func(params, data):
            a, b = params
            n = len(data)
            c = (data - a) / b
            x1 = np.sum(sc.expit(c)) - n/2
            x2 = np.sum(c*np.tanh(c/2)) - n
            return x1, x2
```

Reply:
Thanks for taking a look - I was trying that to see if it would make any difference but realized that in the other equation you wrote for gumbel_r the exp(M)s were able to be cancelled out.

Reply:
I'm going to commit this change, and if the test suite completes successfully, I'll merge the PR.